README.md

```diff
@@ -1,222 +1,515 @@
 ```
+
 **This is okay, too:**
+
 ```javascript
-var foo, bar, baz;
+let foo, bar, baz;
 ```
 
-## Function Declarations vs. Function Expressions
+#### Unused Variables
+
+Unused variables should be removed from the code when they are no longer needed.
 
-Prefer `var`ed function expressions over function declarations.
+Although discouraged, unused function parameters are allowed. This is due to some libraries expecting specific function signatures.
 
 **Don't do this:**
+
 ```javascript
-function() {
-   function foo() { ... };
-}();
+const doSomething = function() {
+  // foo is never used
+  let foo;
+
+  return 'Hello World!';
+}
 ```
-**Do this instead:**
+
+**This is allowed:**
+
 ```javascript
-function() {
-  var foo = function() { ... };
-}();
+const doSomething = function(error, data) {
+  // data is never used
+  if (error) {
+    throw error;
+  }
+});
 ```
 
-## Aligning Assignments
+#### Undeclared Variables
+
+All variables must be explicitly declared.
+
+This helps catch errors due to mistyped variable names and copy & pasting.
+Additionally, it helps prevent dependence on the global state.
+As a result, modules are forced to be explicit about their dependencies.
+
+For example, using an undeclared `$` variable in Backbone views may have unexpected results.
+Using the global jQuery variable (`$`) accidentally, assuming it exists, will result in operations affecting the entire DOM, instead of only affecting the view.
+
+#### Using Variables Before Declaration
+
+Variables must be declared before they are used. Hoisting makes it possible to accidentally use a variable before it's declared.
+
+#### Aligning Assignments
+
+Variable assignments shouldn't be aligned by value.
 
-Variable assignments should not be aligned by value.
 **Don't do this:**
+
 ```javascript
-var foo      = 'foo';
-var bar      = 'bar';
-var fizzbuzz = 'fizzbuzz';
+let foo      = 'foo';
+let bar      = 'bar';
+let fizzbuzz = 'fizzbuzz';
 
-var obj = {
+let obj = {
   foo     : 'foo',
   bar     : 'bar',
   fizzbuzz: 'fizzbuzz'
 };
 ```
+
 **Do this instead:**
+
 ```javascript
-var foo = 'foo';
-var bar = 'bar';
-var fizzbuzz = 'fizzbuzz';
+let foo = 'foo';
+let bar = 'bar';
+let fizzbuzz = 'fizzbuzz';
 
-var obj = {
+let obj = {
   foo: 'foo',
   bar: 'bar',
   fizzbuzz: 'fizzbuzz'
 };
 ```
 
-## Anonymous Callback Functions
+### Constants
+
+This section conflicts with the Google JavaScript Style Guide.
+The following rules should be followed instead.
 
-Prefer `var`ed function expressions over anonymous functions for callbacks where it makes sense.
+* The `@const` tag shouldn't be used.
+* When available, the `const` keyword should be used.
+* Consider [freezing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) object constants when applicable.
+
+### Functions
+
+Function expressions should be used instead of function declarations.
 
 **Don't do this:**
+
+```javascript
+function foo() {
+  /// [...]
+};
```

Comment:
Even if we're not using them, function declarations shouldn't have a semi-colon.